app/questionnaire/questionnaire_manager.py

```diff
@@ -85,8 +86,9 @@ def build_state(self, location, answers):
             metadata = get_metadata(current_user)
             answer_store = get_answer_store(current_user)
             schema_item = self._schema.get_item_by_id(location['block_id'])
-
             self.state = schema_item.construct_state()
+            self.state.first_block = schema_item.id == schema_item.container.blocks[0].id
```

Comment:
Do we need to do it this way? Can we just conditionally set `previous_url` to None in the `render_theme_template` of `questionnaire.py` if we don't want to display the previous url?
Something like `previous_url = None if block_id == SchemaHelper.get_first_block_id_for_group(group_id)`
(new method would need to be added to SchemaHelper)